<launch>
  <arg name="use_robot_name"         default="open_manipulator"/>

  <arg name="use_platform"           default="false"/>
  <arg name="use_moveit"             default="true"/>
  <arg name="planning_group_name"    default="arm"/>

  <arg name="dynamixel_usb_port"     default="/dev/ttyUSB0"/>
  <arg name="dynamixel_baud_rate"    default="1000000"/>

  <arg name="control_period"         default="0.010"/>

  <group if="$(arg use_moveit)">
    <include file="$(find open_manipulator_controller)/launch/open_manipulator_moveit.launch">
      <arg name="sample_duration" value="$(arg control_period)"/>
    </include>
  </group>

  <node name="$(arg use_robot_name)" pkg="open_manipulator_controller" type="open_manipulator_controller" output="screen" args="$(arg dynamixel_usb_port) $(arg dynamixel_baud_rate)">
      <param name="using_platform"       value="$(arg use_platform)"/>
      <param name="using_moveit"         value="$(arg use_moveit)"/>
      <param name="planning_group_name"  value="$(arg planning_group_name)"/>
      <param name="control_period"       value="$(arg control_period)"/>
  </node>
</launch>
